enhancements/clusterGroupUpgradeStatus.md

```diff
@@ -0,0 +1,319 @@
+  - **nonCompliant**: the policy applied to enforce remediation but it did not get compliant.
+  - **compliant**: the policy applied to enforce remediation and it is compliant.
+  - **timeout**: the policy applied to enforce remediation but it does not become compliant during the timeout limits defined in the remediationStrategy.
+
+The cluster state has 2 possible state;
+  - **complete**: if all the policies has a compliant state on the cluster
+  - **inProgress**: if all the policies are in compliant/nonCompliant or notApplied state.
+  - **failed**: if at least 1 policy has timeout state.
+
+
+```
+apiVersion: cluster.open-cluster-management-extension.io/v1beta1
+kind: ClusterGroupUpgrade
+metadata:
+  name: cgu-upgrade-new
+  namespace: default
+spec:
+  enable: true
+  clusters:
+  - spoke2
+  - spoke3
+  - spoke4
+  - spoke5
+  managedPolicies:
+    - policy1-common
+    - policy2-group
+    - policy3-site
+  remediationStrategy:
+    canaries:
+    - prod1
+    maxConcurrency: 2
+    timeout: 240
+status:
+ conditions: 
+   ...
+ canaryClusters:
+ - name: spoke1
+   policies:
+     policy1-common: compliant
+     policy2-group: compliant
+     policy3-site:  compliant
+   state: complete
+ clusters:
+ - name: spoke2
+   policies:
+     policy1-common: compliant
+     policy2-group: nonCompliant
+     policy3-site:  notApplied
+   state: inProgress
+ - name: spoke3
+   policies:
+     policy1-common: compliant
+     policy2-group: timeout
+     policy3-site:  notApplied
+   state: failed
+ - name: spoke4
+   policies:
+     policy1-common: notApplied
+     policy2-group: notApplied
+     policy3-site:  notApplied
+   state: inProgress
+ - name: spoke5
+   policies:
+     policy1-common: notApplied
+     policy2-group: notApplied
+     policy3-site:  notApplied
+   state: inProgress
+```
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### CGU status 
+
+##### 1- copied policies, placement rules and placement binding names
+
+The CGU controller copies the ACM policies defined under the CGU->managedPolicies field then change its remediation to enforce and create a placement rule and placement binding accordingly to the select clusters.
+While storing the copied policies, placement rules and placement binding names in the CGU status is required by the current implementation of the reconcile process however its not mandatory. Having the copied policies names consist of the CGU name and policy index will make it unique for the reconciling loop.
+Moreover, based on the [selective policy enhamcement](https://github.com/open-cluster-management-io/enhancements/pull/57) there is no needs to copies policies for future integration with ACM coming releases.  
+
+##### 2- batches
+
+The CGU status field store the current cluster batch number in order to iterate to the next batch after success/failed of the running batch.  The new proposed CGU status does not require that as all clusters with indies are stored under clusters list field.
```

Comment:
I don't think it makes sense to force the controller to go through the whole list of clusters on each reconcile, just to figure out which ones it should be remediating. Plus, the current batch status field is more for internal use, therefore the requirements are different. Mixing user oriented requirements and the internal requirements (e.g. policy index) in the same field would be difficult to implement, especially for api compatibility reasons.

Reply:
I don't think there is issue with back compatibility we still do batching . can u elaborate more why its difficult  to implement ?